src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -522,6 +527,19 @@ private boolean checkIfCanceled()
             currentStructure = null;
```

Comment:
This method seems like something that should just be overridable instead of a bunch of instanceof sections